src/reference/torii-endpoints.md

```diff
@@ -1,69 +1,167 @@
+
+**Example**:
 
 ```json
-"1"
+200 OK: "1"
 ```
 
+::: info
+
+The API version is retrieved from and is the same as the version of the [genesis block](../guide/configure/genesis.md), which means that at least a minimal subnet of four peers must be running, and the genesis block must already be submitted at the time of the request.
+
+:::
+
 ## Blocks Stream
 
-- **Protocol:** HTTP
-- **Protocol Upgrade:**`WebSocket
-- **Endpoint:** `/block/stream`
+- **Protocols**: `HTTP` upgraded to `WebSocket`
+- **Encoding**: `SCALE`
+- **Endpoint**: `/block/stream`
+
+#### Handshake
+
+Since the `/block/stream` endpoint handles continuous two-way data exchange, a `WebSocket` handshake between the client and server must first be performed to initiate communication with this endpoint.
 
-The client should send a [`BlockSubscriptionRequest`](/reference/data-model-schema#blocksubscriptionrequest) to initiate
-communication after the WebSocket handshake. Then the server sends a
-[`BlockMessage`](/reference/data-model-schema#blockmessage). Messages are SCALE-encoded[^1].
+The first HTTP request to this endpoint requires a standard set of `WebSocket` headers.
 
-Via this endpoint, the client first provides the starting block number (i.e. height) in the subscription request. After
-sending the confirmation message, the server starts streaming all the blocks from the given block number up to the
-current block and continues to stream blocks as they are added to the blockchain.
+**Example**:
+
+```http
+Host: example.com:8000
+Upgrade: websocket
+Connection: Upgrade
+Sec-WebSocket-Key: x3JJHMbDL1EzLkh9GBhXDw==
+Sec-WebSocket-Version: 13
+Sec-WebSocket-Extensions: permessage-deflate, client_max_window_bits
+```
+
+#### Data Exchange
+
+After a successful handshake, the client must send a [`BlockSubscriptionRequest`](/reference/data-model-schema#blocksubscriptionrequest) request with the starting block number provided (i.e., the `height` value). Then, upon sending the confirmation and [`BlockMessage`](/reference/data-model-schema#blockmessage) messages, the server starts streaming all of the blocks, beginning with the block specified with `height` up to the most recent one, and then continues to stream new blocks as they are added to the blockchain.
+
+#### Responses
+
+| Code | Response              | Description                                                                   |
+| :--: | --------------------- | ----------------------------------------------------------------------------- |
+| 101  | Switching Protocols   | The protocol is successfully switched from `HTTP` to `WebSocket`.             |
+| 400  | Bad Request           | The server will not process the request due to an issue on the client side.   |
+| 401  | Unauthorized          | The client lacks valid credentials for the request.                           |
+| 500  | Internal Server Error | The server encountered an unexpected issue and could not fulfill the request. |
```

Comment:
- 101 - might occur, doesn't matter for Torii docs
- 400 - never
- 401 - never, this request doesn't require authorization
- 500 - might occur for any request, isn't worth to mention

I would say, the whole table is not needed, as well as "responses" section.